### PR TITLE
feat: Update dependencies to latest versions

### PR DESCRIPTION
**Pull request by bgruening**

### CODE REVIEW

File: bioconda_utils/bioconda_utils-requirements.txt [hunk 1]
```diff
@@ -7,21 +7,21 @@ conda-forge-pinning=2024.06.01.05.54.15
 
 # basics
 python=3.10.*
```
Comment (bgruening) [inline bioconda_utils/bioconda_utils-requirements.txt:9]: I'm too afraid to bump python to 3.12 ... but if someone says its a good idea, please let me know!

File: bioconda_utils/bioconda_utils-requirements.txt [hunk 2]
```diff
@@ -7,21 +7,21 @@ conda-forge-pinning=2024.06.01.05.54.15
...
 conda-libmamba-solver=24.1.*
-conda-build=24.5.*
-conda-index=0.4.*
-conda-package-streaming=0.9.*
+conda-build=24.9.*
+conda-index=0.5.*
+conda-package-streaming=0.11.*
 mamba=1.5.*
-boa=0.17.*
+#boa=0.17.*
```
Comment (bgruening) [inline bioconda_utils/bioconda_utils-requirements.txt:16]: Are we using this one? This one is currently blocked by https://github.com/conda-forge/boa-feedstock/pull/85

Comment (aliciaaevans) [reply bioconda_utils/bioconda_utils-requirements.txt:16]: Looks like the boa imports were all removed in https://github.com/bioconda/bioconda-utils/pull/980

Comment (bgruening) [reply bioconda_utils/bioconda_utils-requirements.txt:16]:
```suggestion
```

**Pull request closed without merging**